gramineproject/gramine
# [Pal] Remove `pal_sec` and refactor topology structures

## Description of the changes 

This is the core of #314 after splitting out other changes.

This is a preparation for a better topology sanitization and deduplication of PAL code.

## How to test this PR? 

CI + OpenVINO example (quoting @dimakuv: "OpenVINO is known to check sysfs. Without proper sysfs configuration, OpenVINO still works (IIRC) but is very slow."). Also, manual check for CPUID sanitization change.


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/gramineproject/gramine/325)
